File: frontend/src/api/project.ts
import api from './index'

export interface Project {
  id: number
  name: string
  description: string
  status: string
  version: number
  creator_id: number
  created_at: string
  updated_at: string
  members?: ProjectMember[]
}

export interface ProjectMember {
  id: number
  user_id: number
  username: string
  email: string
  role: string
  avatar_url: string
}

export interface ProjectListResponse {
  list: Project[]
  total: number
  page: number
  page_size: number
}

export function createProject(data: { name: string; description: string }) {
  return api.post<Project>('/api/projects', data)
}

export function getProjects(params?: { page?: number; page_size?: number }) {
  return api.get<ProjectListResponse>('/api/projects', { params })
}

export function getProject(id: number | string) {
  return api.get<Project>(`/api/projects/${id}`)
}

export function updateProject(id: number | string, data: { name: string; description: string; version: number }) {
  return api.put(`/api/projects/${id}`, data)
}

export function submitProject(id: number | string) {
  return api.post(`/api/projects/${id}/submit`)
}

export function approveProject(id: number | string) {
  return api.post(`/api/projects/${id}/approve`)
}

export function startProject(id: number | string) {
  return api.post(`/api/projects/${id}/start`)
}

export function completeProject(id: number | string) {
  return api.post(`/api/projects/${id}/complete`)
}

export function archiveProject(id: number | string) {
  return api.post(`/api/projects/${id}/archive`)
}

export function addMember(projectId: number | string, data: { user_id: number }) {
  return api.post(`/api/projects/${projectId}/members`, data)
}

export function removeMember(projectId: number | string, userId: number) {
  return api.delete(`/api/projects/${projectId}/members/${userId}`)
}

export function getMembers(projectId: number | string) {
  return api.get<ProjectMember[]>(`/api/projects/${projectId}/members`)
}

File: backend/internal/model/project.go
package model

import "time"

type Project struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	Name        string    `gorm:"size:200;not null" json:"name"`
	Description string    `gorm:"type:text" json:"description"`
	Status      string    `gorm:"size:30;default:Draft;not null" json:"status"`
	OwnerID     uint      `gorm:"not null;index" json:"owner_id"`
	Owner       *User     `gorm:"foreignKey:OwnerID" json:"owner,omitempty"`
	Version     int       `gorm:"default:1" json:"version"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type ProjectMember struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	ProjectID uint      `gorm:"not null;index" json:"project_id"`
	UserID    uint      `gorm:"not null;index" json:"user_id"`
	User      *User     `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Role      string    `gorm:"size:20;default:Developer" json:"role"`
	JoinedAt  time.Time `gorm:"autoCreateTime" json:"joined_at"`
}
